### PR TITLE
docs: modify `architecture-design.md` to add what should pay attention when…

### PR DESCRIPTION
**Pull request by idbeta**
fixed https://github.com/apache/apisix/issues/2741

### CODE REVIEW

**Review by spacewander — APPROVED** [review]

File: doc/architecture-design.md [hunk 1]
```diff
@@ -188,7 +188,7 @@ Note: When both Route and Service enable the same plugin, the Route parameter ha
 
 The `Plugin` configuration can be bound directly to `Route` or it can be bound to `Service` or `Consumer`. For the configuration of the same plugin, only one copy is valid, and the configuration selection priority is always `Consumer` > `Route` > `Service`.
 
-In `conf/config.yaml`, you can declare which plugins are supported by the local APISIX node. This is a whitelisting mechanism. Plugins that are not in this whitelist will be automatically ignored. This feature can be used to temporarily turn off or turn on specific plugins, which is very effective in dealing with unexpected situations.
+In `conf/config.yaml`, you can declare which plugins are supported by the local APISIX node. This is a whitelisting mechanism. Plugins that are not in this whitelist will be automatically ignored. This feature can be used to temporarily turn off or turn on specific plugins, which is very effective in dealing with unexpected situations. If you want to add new plugins based on existing plugins, you need to copy the `plugins` node content of `config-default.yaml` to the `plugins` node of `conf/config.yaml`.
```
Comment (Firstsawyou) [inline doc/architecture-design.md:191]: I think `config-default.yaml` is changed to `conf/config-default.yaml`. This is better, keep it consistent with `conf/config.yaml`, and also indicate in which directory the `config-default.yaml` file is located. What do you think?

File: doc/zh-cn/architecture-design.md [hunk 2]
```diff
@@ -194,6 +194,7 @@ curl http://127.0.0.1:9080/apisix/admin/routes/102 -H 'X-API-KEY: edd1c9f034335f
 
 在 `conf/config.yaml` 中，可以声明本地 APISIX 节点都支持哪些插件。这是个白名单机制，不在该
 白名单的插件配置，都将会被自动忽略。这个特性可用于临时关闭或打开特定插件，应对突发情况非常有效。
+如果你想在现有插件的基础上新增插件，注意需要拷贝 `config-default.yaml` 的插件节点内容到 `conf/config.yaml` 的插件节点中。
```
Comment (Firstsawyou) [inline doc/zh-cn/architecture-design.md:197]: ditto.

File: doc/architecture-design.md (hunk 1 above)
Comment (moonming) [inline doc/architecture-design.md:191]: why we need copy configure file? @membphis

Comment (idbeta) [reply doc/architecture-design.md]: I agree, fixed.

Comment (membphis) [reply doc/architecture-design.md]: we do not need to `copy` now. we can remove this step

Comment (membphis) [reply doc/architecture-design.md]: @idbeta I do not think you have fixed, please check this

Comment (moonming) [reply doc/architecture-design.md]: @idbeta any update?

Comment (idbeta) [reply doc/architecture-design.md]: I have updated it, you can check again. @moonming @membphis

**Review by membphis — APPROVED** [review]

**Pull request merged**